### PR TITLE
Feat: Migrate to Jest

### PR DESCRIPTION
**Pull request by sseppola**
Replaces Ava with Jest to align our testing frameworks.

With many dependency changes the diff looks huge, but it's mostly in yarn.lock:
Screenshot 2023-03-10 at 11 45 46

Some types were also moved in order to make consistent coverage ignore patterns.

### CODE REVIEW

File: jest-coverage.config.mjs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  collectCoverage: true,
+  collectCoverageFrom: [
+    './src/**/*',
+    '!./src/types/*',
+    '!./src/**/types.ts',
+    '!./src/**/index.ts',
+    '!./src/**/*.tspec.ts',
+  ],
+  coverageThreshold: {
+    global: { statements: 95, branches: 95 },
```
Comment (AlicjaMajewska) [inline jest-coverage.config.mjs:14]: that's a high coverage, noice 👌

File: yarn.lock [hunk 2]
```diff
@@ -72,11 +293,725 @@
...
+  version "7.21.2"
+  resolved "https://registry.yarnpkg.com/@babel/parser/-/parser-7.21.2.tgz#dacafadfc6d7654c3051a66d6fe55b6cb2f2a0b3"
+  integrity sha512-URpaIJQwEkEC2T9Kn+Ai6Xe/02iNaVCuT/PtoRz3GPVJVDpPd7mLo+VddTbhCRU9TXqW5mSrQfXZyi8kDKOVpQ==
+
 "@babel/parser@^7.10.3", "@babel/parser@^7.18.10", "@babel/parser@^7.18.13":
   version "7.18.13"
   resolved "https://registry.yarnpkg.com/@babel/parser/-/parser-7.18.13.tgz#5b2dd21cae4a2c5145f1fbd8ca103f9313d3b7e4"
   integrity sha512-dgXcIfMuQ0kgzLB2b9tRZs7TTFFaGM2AbtA4fJgUUYukzGH4jwsS7hzQHEGs67jdehpm22vkgKwvbU+aEflgwg==
 
+"@babel/plugin-bugfix-safari-id-destructuring-collision-in-function-expression@^7.16.2":
```
Comment (AlicjaMajewska) [inline yarn.lock:306]: this explosion of babel plugins is crazy 🤯

File: src/persistentReducedStream.spec.ts [hunk 3]
```diff
@@ -167,14 +169,14 @@ test(
         state$.unsubscribe();
         try {
           state$.connect();
-          t.fail('connect should throw an error');
+          fail('connect should throw an error');
```
Comment (AlicjaMajewska) [inline src/persistentReducedStream.spec.ts:172]: I would replace it with expect(....).toThrow(), but we can also keep as is

**Review by AlicjaMajewska — APPROVED** [review]
LGTM

File: yarn.lock (hunk 2 above)
Comment (holographic-principle) [reply yarn.lock:306]: :rofl:

**Pull request merged**